src/components/draw/DrawDirective.js

```diff
@@ -27,889 +28,798 @@ goog.require('ga_permalink');
+          // Remove help overlay
+          if (helpTooltip) {
+            map.removeOverlay(helpTooltip);
+          }
 
-              // Add temporary measure tooltips
-              if (tool.showMeasure) {
-                // Distance overlay
-                distTooltip = createMeasureTooltip();
-                map.addOverlay(distTooltip);
+        };
 
-                // Area overlay
-                areaTooltip = createMeasureTooltip();
-                map.addOverlay(areaTooltip);
-              }
+        // Set the draw interaction with the good geometry
+        var activateDrawInteraction = function(tool) {
+          select.setActive(false);
+          deactivateDrawInteraction();
 
-              deregFeatureChange = evt.feature.on('change', function(evt) {
-                var geom = evt.target.getGeometry();
-                if (geom instanceof ol.geom.Polygon) {
-                  var lineCoords = geom.getCoordinates()[0];
-
-                  if (nbPoint != lineCoords.length) {
-                    // A point is added
-                    nbPoint++;
-                  } else {
-                    var firstPoint = lineCoords[0];
-
-                    // We update features and measures
-                    var lastPoint = lineCoords[lineCoords.length - 1];
-                    var lastPoint2 = lineCoords[lineCoords.length - 2];
-
-                    var isSnapOnFirstPoint = (lastPoint[0] == firstPoint[0] &&
-                        lastPoint[1] == firstPoint[1]);
-
-                    // When the last change event is triggered the polygon is
-                    // closed so isSnapOnFirstPoint is true. We need to know
-                    // if on the change event just before, the snap on last
-                    // point was active.
-                    isFinishOnFirstPoint = (!isSnapOnLastPoint &&
-                        isSnapOnFirstPoint);
-
-                    isSnapOnLastPoint = (lastPoint[0] == lastPoint2[0] &&
-                        lastPoint[1] == lastPoint2[1]);
-
-                    if (isSnapOnLastPoint) {
-                      // In that case the 2 last points of the coordinates
-                      // array are identical, so we remove the useless one.
-                      lineCoords.pop();
-                    }
-                  }
-                  updateHelpTooltip(helpTooltip, tool.id, true,
-                      isFinishOnFirstPoint, isSnapOnLastPoint);
-                  if (tool.showMeasure) {
-                    if (!isFinishOnFirstPoint) {
-                      geom = new ol.geom.LineString(lineCoords);
-                    }
-                    updateMeasureTooltips(distTooltip, areaTooltip, geom);
-                  }
-                }
-              });
+          updateHelpTooltip(helpTooltip, tool.id, false);
+
+          if (!gaBrowserSniffer.mobile) {
+            unDrawEvts.push(map.on('pointermove', function(evt) {
+              helpTooltip.setPosition(evt.coordinate);
```

Comment:
We have the same event binding in the select change:active function. Not sure we wheter we need it in both places.

If yes, might be factorized.